### PR TITLE
feat: create social authentication guide

### PR DESCRIPTION
**Pull request by AtofStryker**
Creates the Social Auth guide for examples with Google, Facebook, and Microsoft as well as our recommendations around them. The point of this guide is to give users a path forward who need to test these flows, even though we do not recommend implementing them in CI or having your tests primarily rely on them

~I have some videos being checked in. I am working with @elylucas to get these moved over to vimeo along with the other video content from the v12 update. One of the videos has a watermark. I will work on getting rid of it tomorrow as I get the media updated, but the rest of the content I believe is ready for review~ UPDATE: All videos do not have watermarks and are moved to vimeo

### CODE REVIEW

File: content/guides/end-to-end-testing/social-authentication.md [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+
+  it('navigates to the app after navigation and displays the sample project header', () => {
+    cy.get('h1').should('contain', 'React.js Sample Project')
+  })
+})
+```
+
+Here is a sample of what all 3 should look like in order.
+
+TODO: remove watermark
```
Comment (AtofStryker) [inline content/guides/end-to-end-testing/social-authentication.md:337]: Need to remake this video as well as upload the videos to vimeo. The vimeo update might be done in a separate PR

File: content/guides/end-to-end-testing/social-authentication.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+  "FACEBOOK_PASSWORD": "",
+  "FACEBOOK_NAME": ""
+}
+```
+
+### Login with [`cy.origin()`](/api/commands/origin)
+
+Next, we'll write a custom command called `loginToAuth0ViaSocial` to perform a
+login to either Facebook, Google, or Microsoft. This command will use
+[`cy.origin()`](/api/commands/origin) to
```
Comment (mjhenkes) [inline content/guides/end-to-end-testing/social-authentication.md:133]:
```suggestion
Next, we'll write a custom command called `loginToAuth0ViaSocial` to perform a
login to either Facebook, Google, or Microsoft. This command will use
[`cy.origin()`](/api/commands/origin) to:
```

**Review by mjhenkes — APPROVED** [review]

File: content/guides/end-to-end-testing/social-authentication.md (hunk 2 above)
Comment (AtofStryker) [reply content/guides/end-to-end-testing/social-authentication.md:133]: addressed in 822876bea

File: content/guides/end-to-end-testing/social-authentication.md (hunk 1 above)
Comment (AtofStryker) [reply content/guides/end-to-end-testing/social-authentication.md]: addressed in 894595edb

File: content/guides/end-to-end-testing/social-authentication.md (hunk 2 above)
Comment (AtofStryker) [reply content/guides/end-to-end-testing/social-authentication.md]: addressed in 822876bea

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+
+:::info
+
+## <Icon name="graduation-cap"></Icon> What you'll learn
+
+- Log into your application via Google, Facebook, or Microsoft through the UI
+  with [`cy.origin()`](/api/commands/origin) using Auth0 social connections. The
+  same patterns will work with Github, LinkedIn, and others.
+- Understanding
+  [challenges](http://localhost:3000/guides/references/best-practices#When-logging-in)
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:14]: should this be a localhost link?

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+to avoid A/B tests or alternate login flows. Relying on social authentication in
+CI will likely result in bot detection and in some cases, account suspension due
+to violating the provider's Terms of Service.
+
+:::
+
+## Application Setup
+
+For this guide, we are going to set up an application that uses Auth0 for
+authentication. Our Auth0 Tenant will have social connections set up for Google,
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:45]:
```suggestion
authentication. Our Auth0 tenant will have social connections set up for Google,
```

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+Facebook, and Microsoft which with authenticate users on our behalf to our
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:46]:
```suggestion
Facebook, and Microsoft which will authenticate users on our behalf to our
```

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+  - Step 1: When configuring your Sample App, please select the
+    `Single-Page App` for you platform and`React` for the technology.
+  - Step 2: For `Social Connections`, please select Facebook, Microsoft, and
+    Google
+  - Step 3: Click try login and verify the connections are set up properly
+  - Step 4: You can either download the sample on the `Download Sample App`
+    button or clone it
+    [here](https://github.com/auth0-samples/auth0-react-samples/tree/d5848ed710db85d6ef060731b296603c0eca744b/Sample-01).
+    We will need this application in the next step. Select
+    `"I am Ready to Use Auth0"`. Your social connections on your Auth0 Tenant
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:62]:
```suggestion
    `"I am Ready to Use Auth0"`. Your social connections on your Auth0 tenant
```

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+:::
+
+## Application Setup
+
+For this guide, we are going to set up an application that uses Auth0 for
+authentication. Our Auth0 Tenant will have social connections set up for Google,
+Facebook, and Microsoft which with authenticate users on our behalf to our
+application.
+
+### Setting up the Tenant
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:49]:
```suggestion
### Setting up the Auth0 tenant
```

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+    Google
+  - Step 3: Click try login and verify the connections are set up properly
+  - Step 4: You can either download the sample on the `Download Sample App`
+    button or clone it
+    [here](https://github.com/auth0-samples/auth0-react-samples/tree/d5848ed710db85d6ef060731b296603c0eca744b/Sample-01).
+    We will need this application in the next step. Select
+    `"I am Ready to Use Auth0"`. Your social connections on your Auth0 Tenant
+    should now be configured correctly
+
+### Setting up the Application
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:65]:
```suggestion
### Setting up the application
```

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 9]
```diff
@@ -0,0 +1,423 @@
...
+  file as described in the
+  [sample app](https://github.com/auth0-samples/auth0-react-samples/tree/d5848ed710db85d6ef060731b296603c0eca744b/Sample-01#create-an-api).
+- Once created, paste the `domain` and `client ID` into their respective fields
+  inside `src/auth_config.json`. The `audience` key can be removed. The
+  login/callback URIs will default to `http://localhost:3000` which is the url
+  of our test application.
+- [install](https://github.com/auth0-samples/auth0-react-samples/tree/d5848ed710db85d6ef060731b296603c0eca744b/Sample-01#project-setup)
+  the application dependencies and
+  [start the dev server](https://github.com/auth0-samples/auth0-react-samples/tree/d5848ed710db85d6ef060731b296603c0eca744b/Sample-01#compile-and-hot-reload-for-development)
+- Verify the application is stood up correctly by visiting
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:89]:
```suggestion
- Verify the application is running correctly by visiting
```

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 10]
```diff
@@ -0,0 +1,423 @@
...
+  finished, click the `Log Out` dropdown button. Repeat for Google, Microsoft,
+  and Facebook or the accounts you would like to test.
+
+Here is what this process should look like for Facebook setup:
+
+<DocsVideo src="https://vimeo.com/789093604"></DocsVideo>
+
+## Testing with Cypress
+
+Now that our application is set up and ready to go, we are ready to write our
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:103]: Does this section need more attention drawn to it?  Iconography or subheading perhaps?

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 11]
```diff
@@ -0,0 +1,423 @@
...
+      cy.get('input[type="submit"]').click()
+      cy.get('#idBtn_Back').click()
+    }
+  )
+
+  cy.get('h6.dropdown-header').should('contain', name)
+}
+```
+
+We can wire together these functions in a single command, like so:
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:272]:
```suggestion
We can wire these functions together in a single command, like so:
```

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx [hunk 12]
```diff
@@ -0,0 +1,423 @@
...
+    }
+
+    log.snapshot('after')
+    log.end()
+  }
+)
+```
+
+Now, we can use our `loginToAuth0ViaSocial` command in the test. Below is our
+test to login as a user via Auth0 and run a basic sanity check.
```
Comment (nagash77) [inline docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:321]:
```suggestion
test to login as a user via Auth0 and run a basic check.
```

**Review by nagash77 — CHANGES_REQUESTED** [review]
Left some readability changes.  There is a broken link I believe as noted.  I also think this could use a bit of a wrap up at the bottom with links to other articles to further the users education.  

Overall this is an excellent guide.  Nice work Bill!

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 5 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in ac6af55da

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 6 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in ac6af55da

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 8 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in ac6af55da

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 9 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in ac6af55da

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 11 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in ac6af55da

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 12 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in ac6af55da

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 4 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in ac6af55da

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 10 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx:101]: I am not sure. Usually with our other guides we tend to go with the H2 here, but I feel like in the others as well that the sections that matter somewhat get swallowed

File: docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx (hunk 7 above)
Comment (AtofStryker) [reply docs/guides/end-to-end-testing/testing-strategies/social-authentication.mdx]: addressed in https://github.com/cypress-io/cypress-documentation/commit/ac6af55da2402a2a483d092323c97b44e6086712

File: docs/guides/references/best-practices.mdx [hunk 13]
```diff
@@ -327,6 +327,8 @@ are common for:
 - CMS instances, such as a Contentful or Wordpress instance.
 - Other types of services under a domain in which you control.
 
+#### Challenges with Social Connection Authentication
```
Comment (nagash77) [inline docs/guides/references/best-practices.mdx:330]: I'm not sure how I feel about Social Connection Authentication.  

```suggestion
#### Potential Challenges Authenticating with Social Platforms
```

**Review by nagash77 — APPROVED** [review]

File: docs/guides/references/best-practices.mdx (hunk 13 above)
Comment (AtofStryker) [reply docs/guides/references/best-practices.mdx]: suggestion reads better imo
